src/commands/registry.rs

```diff
@@ -92,8 +112,27 @@ impl Pull {
     /// Pull a Spin application from an OCI registry
     pub async fn run(self) -> Result<()> {
         let mut client = spin_oci::Client::new(self.insecure, None).await?;
-        client.pull(&self.reference).await?;
 
+        let pb = ProgressBar::new_spinner();
+        pb.enable_steady_tick(Duration::from_millis(200));
+        pb.set_style(
+            ProgressStyle::with_template("{msg}{spinner}")
+                .unwrap()
+                .tick_strings(&[".", "..", "...", "....", "....."]),
+        );
+        pb.set_message("Pulling app from registry");
+
+        let result = client.pull(&self.reference).await;
+        match result {
+            Ok(_) => {
+                pb.set_style(ProgressStyle::with_template("{msg:.green}")?);
+                pb.finish_with_message("Successfully pulled app from registry");
+            }
+            _ => {
```

Comment:
This looks like we are now discarding the error rather than reporting it back to the user?

Reply:
Will revert this.